src/proto/beats.proto

```diff
@@ -10,22 +11,32 @@ enum Source {
   USER = 2;
 }
 
-message Beat {
-  optional int32 frame_position = 1;
-  optional bool enabled = 2 [ default = true ];
-  optional Source source = 3 [ default = ANALYZER ];
+enum Type {
+  BEAT = 0;
+  BAR = 1;
+  PHRASE = 2;
+}
+
+message TimeSignature {
+  required int32 beats_per_bar = 1 [ default = 4];
+  required int32 note_value = 2 [ default = 4 ];
 }
 
 message Bpm {
-  optional double bpm = 1;
+  required double bpm = 1;
   optional Source source = 2 [ default = ANALYZER ];
 }
 
-message BeatMap {
-  repeated Beat beat = 1;
+message Beat {
+  // The frame position should always be explicitly set
+  // otherwise serialization would lead to SIGABRT.
+  required double frame_position = 1 [ default = 0 ];
```

Comment:
Changing the type from int32 to double will be wire-format incompatible with previous Mixxx libraries. Is this proto going to be used to parse protos from previous version of Mixxx?

Reply:
Yes. the format will be broken when read from the serialized representation. It will only be updated when new data is generated by the analyzer thread.

Reply:
Hm, what's the plan for migrating user data from previous versions of Mixxx?